src/nvim/strings.c

```diff
@@ -45,6 +45,41 @@
 #include "nvim/os/os.h"
 #include "nvim/os/shell.h"
 
+/// Compares two strings for equality.
+bool strequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcmp(a, b) == 0;
+}
+
+/// Compares two sized strings for equality.
+bool strnequal(const char *a, const char *b, size_t size)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strncmp(a, b, size) == 0;
+}
+
+/// Compares two strings for equality ignoring differences in case.
+bool striequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcasecmp(a, b) == 0;
```

Comment:
So best idea is splitting them into a) ascii_str\*cmp (used in most cases, including `==#`/`!=#`(these operators are used to make sure that strings are (not) identical) and `[<>]=?#`) and b) something based on ICU for `[<>]=?\?` and `[!=]=\?` and rare occasions (if any) in the C code, also for `strcmp()` VimL function because `#` variants all use byte comparisons (for compatibility with plugins which use strings as byte strings: there are some)?